tests/topotests/bgp-snmp-mplsl3vpn/test_bgp_snmp_mplsvpn.py

```diff
@@ -0,0 +1,730 @@
+
+    # take Last changed time, fiddle with associated interfaces, ensure
+    # time changes and active interfaces change
+    ts_last = r1_snmp.get(
+        "mplsL3VpnVrfConfLastChanged.{}".format(snmp_str_to_oid("VRF-a"))
+    )
+    ts_val_last_1 = get_timetick_val(ts_last)
+    r1.cmd("ip link set r1-eth6 master VRF-a")
+    r1.cmd("ip link set r1-eth6 up")
+
+    associated_int = r1_snmp.get(
+        "mplsL3VpnVrfAssociatedInterfaces.{}".format(snmp_str_to_oid("VRF-a"))
+    )
+    assertmsg = "mplsL3VpnVrfAssociatedInterfaces incorrect should be 3 value {}".format(
+        associated_int
+    )
+
+    assert associated_int == "3", assertmsg
+    ts_last = r1_snmp.get(
+        "mplsL3VpnVrfConfLastChanged.{}".format(snmp_str_to_oid("VRF-a"))
+    )
+    ts_val_last_2 = get_timetick_val(ts_last)
+    assertmsg = "mplsL3VpnVrfConfLastChanged does not update on interface change"
+    assert ts_val_last_2 > ts_val_last_1, assertmsg
+    r1.cmd("ip link del r1-eth6 master VRF-a")
+    r1.cmd("ip link set r1-eth6 down")
+
+
+rt_table_test = {
+    "mplsL3VpnVrfRT": ['"1:1"', '"1:2"'],
+    "mplsL3VpnVrfRTDescr": ["RT both for VRF VRF-a", "RT both for VRF VRF-b"],
+    "mplsL3VpnVrfRTRowStatus": ["active(1)", "active(1)"],
+    "mplsL3VpnVrfRTStorageType": ["volatile(2)", "volatile(2)"],
+}
+
+
+def test_r1_mplsvpn_VrfRT_table():
+    tgen = get_topogen()
+
+    r1 = tgen.net.get("r1")
+    r1r = tgen.gears["r1"]
+
+    r1_snmp = SnmpTester(r1, "1.1.1.1", "public", "2c")
+
+    oids = []
+    oids.append(generate_vrf_index_type_oid("VRF-a", 1, 3))
+    oids.append(generate_vrf_index_type_oid("VRF-b", 1, 3))
+
+    # check items
+    for item in rt_table_test.keys():
+        print(item)
+        assertmsg = "{} should be {} oids {} full dict {}:".format(
+            item, rt_table_test[item], oids, r1_snmp.walk(item)
+        )
+        assert r1_snmp.test_oid_walk(item, rt_table_test[item], oids), assertmsg
+
+
+def test_r1_mplsvpn_perf_table():
+    tgen = get_topogen()
+
+    r1 = tgen.net.get("r1")
+    r1r = tgen.gears["r1"]
+
+    r1_snmp = SnmpTester(r1, "1.1.1.1", "public", "2c")
+
+    # tgen.mininet_cli()
+    oid_a = snmp_str_to_oid("VRF-a")
+    oid_b = snmp_str_to_oid("VRF-b")
+
+    # poll for 10 seconds for routes to appear
+    count = 0
+    passed = False
+    while count < 60:
+        if r1_snmp.test_oid_walk(
+            "mplsL3VpnVrfPerfCurrNumRoutes.{}".format(oid_a), ["7"]
+        ):
+            passed = True
+            break
+        count += 1
+        sleep(1)
+    # tgen.mininet_cli()
+    assert passed
+    curr_a = int(r1_snmp.get("mplsL3VpnVrfPerfCurrNumRoutes.{}".format(oid_a)))
+    del_a = int(r1_snmp.get("mplsL3VpnVrfPerfRoutesDeleted.{}".format(oid_a)))
+    add_a = int(r1_snmp.get("mplsL3VpnVrfPerfRoutesAdded.{}".format(oid_a)))
+
+    assertmsg = "FAIL curr{} does not equal added{} - deleted {}".format(
+        curr_a, add_a, del_a
+    )
+    assert curr_a == (add_a - del_a)
```

Comment:
Did you miss the `assertmsg` as defined above? It's not added to the `assert` command